Pull request in AzureDevops is not working when source is external

see https://dev.azure.com/cmunckhof/Open%20Source/_build/results?buildId=1987&view=results

Run analysis error message

> 
> Starting: Run Code Analysis
> ==============================================================================
> Task         : Run Code Analysis
> Description  : Run scanner and upload the results to the SonarCloud server.
> Version      : 1.45.0
> Author       : sonarsource
> Help         : Version: 1.45.0. This task is not needed for Maven and Gradle projects since the scanner should be run as part of the build.
> 
> [More Information](https://docs.sonarcloud.io/advanced-setup/ci-based-analysis/sonarcloud-extension-for-azure-devops/)
> ==============================================================================
> ##[error]Variables are missing. Please make sure that you are running the Prepare task before running the Analyze task.
> Make sure you are not mixing tasks from different major versions. If you are using a multistage pipeline, make sure the tasks are in the same stage.
> Finishing: Run Code Analysis

--- a/.azuredevops/Pipelines/Templates/prepare-code-analysis.yaml
+++ b/.azuredevops/Pipelines/Templates/prepare-code-analysis.yaml
@@ -10,3 +10,5 @@ steps:
     extraProperties: |
       sonar.cs.vstest.reportsPaths=$(Agent.TempDirectory)/**/*.trx
       sonar.cs.opencover.reportsPaths=$(Agent.TempDirectory)/**/*.opencover.xml
+  condition: or( ne(variables['Build.Reason'], 'PullRequest'), eq(variables['System.PullRequest.IsFork'], 'false') )
+  

--- a/.azuredevops/Pipelines/Templates/publish-code-analysis.yaml
+++ b/.azuredevops/Pipelines/Templates/publish-code-analysis.yaml
@@ -2,9 +2,11 @@ steps:
 
 - task: SonarCloudAnalyze@2
   displayName: ⚙️ Run Code Analysis
+  condition: or( ne(variables['Build.Reason'], 'PullRequest'), eq(variables['System.PullRequest.IsFork'], 'false') )
 
 - task: SonarCloudPublish@2
   displayName: 📢 Publish Quality Gate Result
+  condition: or( ne(variables['Build.Reason'], 'PullRequest'), eq(variables['System.PullRequest.IsFork'], 'false') )
 
 - pwsh: |
     dotnet tool install --tool-path . dotnet-reportgenerator-globaltool

